src/collector/ServicesClient.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { retry } from '@lifeomic/attempt';
+import nodeFetch, { Request } from 'node-fetch';
+
+import { retryableRequestError, fatalRequestError } from './error';
+import { URLSearchParams } from 'url';
+
+export interface ServicesClientInput {
+  apiToken: string;
+}
+
+interface QueryParam {
+  [param: string]: string | string[];
+}
+
+const BASE_URL = 'https://api.bugcrowd.com/';
+
+/**
+ * Services Api
+ * https://docs.bugcrowd.com/reference
+ */
+export class ServicesClient {
+  readonly baseUrl: string;
+  readonly authHeader: { [key: string]: string };
+
+  constructor(config: ServicesClientInput) {
+    this.baseUrl = BASE_URL;
+    this.authHeader = {
+      Authorization: `Token ${config.apiToken}`,
+      Accept: 'application/vnd.bugcrowd+json',
+    };
+  }
+
+  test(): Promise<any[]> {
+    return this.fetch('bounties', { offset: '0', limit: '1' });
+  }
+
+  getBounties(): Promise<any> {
```

Comment:
I'm super happy to see there are no hand-coded types for the third party API responses!

There are GitHub Actions warning around all these `get*` functions because they return `any`. This is they reason for the Opaque types like https://github.com/JupiterOne/graph-trend-micro/blob/master/src/provider/types.ts. They essentially declare to the type system that you totally know what you're doing 😄

Reply:
i'm still torn on this.  On one hand, it's nice not to hard code the provider types, but I also find it helpful to know I'm getting the correct properties by having the types.

Reply:
Yeah, I totally understand the discomfort! There is certainly value in having the types, though I think it less than the value in having tests that hit the APIs and assert the data is converted into the expected shape.

TypeScript types share these properties of unit tests (among others I'm sure):

* There is a computer verifying things are as we expect.
* We can expect the wrong thing, which matches the implementation, so the verification passes but is still wrong.
* Verification only occurs during development, not at runtime.

As a side note, this last point is what bothers me the most about TypeScript - so much effort, no runtime value.

Our goals are to increase confidence and reduce bugs. Having both types *and* tests strengthens that even more, to be sure, but I look at it this way: if there are types provided by the API client, great, but we still need tests since those nearly always make all properties optional and often that is because the API endpoint is dynamic in its responses. When there are no provided types, there is not enough value beyond the tests to warrant maintaining them, as they'll likely be incomplete (since we just add what we have to to make TS happy) and I'll still have to go read the docs to figure out what is available. Also, the more we use `convertProperties`, the less useful that maintenance activity becomes.